api/v1/controllers/utils.js

```diff
@@ -44,6 +45,44 @@ function hasDuplicates(tagsArr) {
   return false;
 }
 
+/**
+ * Check if read only field exists in given object
+ * @param  {String} field - Field name
+ * @param  {Object} obj - Request object
+ * @throws {Object} - Throws validation error is field exists
+ */
+function checkReadOnlyFieldInObj(field, obj) {
+  if (obj.hasOwnProperty(field)) {
+    throw new apiErrors.ValidationError(
+      { explanation: `You cannot modify the read-only field: ${field}` }
+    );
+  }
+}
+
+/**
+ * Throws a validation error if the read-only fields are found in the request.
+ * @param  {Object} req - The request object
+ * @param  {Object} props - The module containing the properties of the
+ *  resource type.
+ */
+function noReadOnlyFieldsInReq(req, props) {
+  const requestBody = req.body;
+  if (props.readOnlyFields) {
+    props.readOnlyFields.forEach((field) => {
+      // if request body is an array, check each object in array.
+      if (typeof requestBody !== 'undefined' &&
+            requestBody && requestBody.constructor === Array) {
```

Comment:
Yes, it is and I think I didn't use it because is only supported in ES5 and both have similar performance. But I guess it does not matter because we are already using ES6 features and I agree Array.isArray is concise and readable.
Let me change this.